packages/backend/server/src/plugins/indexer/tables/block.ts

```diff
@@ -0,0 +1,143 @@
+import { z } from 'zod';
+
+export const BlockSchema = z.object({
+  workspace_id: z.string(),
+  doc_id: z.string(),
+  block_id: z.string(),
+  content: z.union([z.string(), z.string().array()]),
+  flavour: z.string(),
+  blob: z.union([z.string(), z.string().array()]).optional(),
+  ref_doc_id: z.union([z.string(), z.string().array()]).optional(),
+  ref: z.union([z.string(), z.string().array()]).optional(),
+  parent_flavour: z.string().optional(),
+  parent_block_id: z.string().optional(),
+  additional: z.string().optional(),
+  markdown_preview: z.string().optional(),
+  created_by_user_id: z.string(),
+  updated_by_user_id: z.string(),
+  created_at: z.date(),
+  updated_at: z.date(),
+});
+
+export type Block = z.input<typeof BlockSchema>;
+
+export function getBlockUniqueId(block: Block) {
+  return `${block.workspace_id}/${block.doc_id}/${block.block_id}`;
+}
+
+export const blockMapping = {
+  settings: {
+    analysis: {
+      analyzer: {
+        uax_url_email_cjk_bigram: {
+          tokenizer: 'uax_url_email_1024',
+          filter: ['lowercase', 'cjk_bigram', 'unique'],
+        },
+        autocomplete: {
+          tokenizer: 'autocomplete_tokenizer',
+          filter: ['lowercase'],
+        },
+      },
+      tokenizer: {
+        uax_url_email_1024: {
+          type: 'uax_url_email',
```

Comment:
Exact matching URLs and emails is not search-friendly, for example, searching for `toeverything.info` will not be able to search for related emails and URLs, or using standard sharding + cjk_bigram is enough